app/build.gradle.kts

```diff
@@ -233,6 +235,21 @@ dependencies {
     implementation("com.github.Blatzar:NiceHttp:0.4.5") // HTTP Lib
 }
 
+tasks.register<Exec>("retrieveCommitHash") {
+    // task needed because configurationCache does not support parallel git commands
+    group = "build"
+    description = "Retrieves the commit hash."
+
+    doFirst {
+        commandLine("git", "rev-parse", "--short", "HEAD")
+        val commitOutput = layout.projectDirectory.file("commit-hash.txt")
```

Comment:
Would it be possible to move this to the build dir?
Having it as a part of the app code will be confusing, especially with future pull requests where this file might give merge conflicts.
Otherwise this looks very good :pray: :pray: :pray:

Reply:
ok , ill check, currently i am finishing off biometric authenticator now.